### PR TITLE
Results table modal

### LINKED ISSUE #62: Add modal or notification if patient has a positive finding and there is only one patient with that family number
So while we're updating the data model changes we're going to simplify this:

- From the list view we want to pull out the `Patient.identifier` that has use: `Identifier.UseEnum.Official` and store this for use later. This is currently the MRN, but we will update it later to use the family number
- when you click on a specific patient, query the fhir server for all patients which have that identifier. e.g. : `/Patient?identifier=969977`
- if there is only one, then show this modal
Text for modal something like:
```
this patient has a known pathogenic variant,  please offer cascade testing to family. Currently this is the only family member with a test result
```

### PR DESCRIPTION
**Pull request by jhughes982**
- Added modal to display check for cascade testing,
- Also added fallback modals in case there are no patients to iterate through,
- Feedback welcome on the use of guard clauses to prevent undefined data... Is there a better way of doing this?
- closes #62

### CODE REVIEW

File: src/components/results-list/ResultsDataFetcher.tsx [hunk 1]
```diff
@@ -71,16 +72,29 @@ const ResultsDataFetcher: FC = () => {
...
     patients.forEach((patient) => {
-      if (!patient.id || !patient.name || !patient.name[0].given) {
+      if (
+        !patient.id ||
+        !patient.identifier ||
+        !patient.identifier[0].value ||
+        !patient.name ||
+        !patient.name[0].given ||
+        !patient.name[0].family
+      ) {
```
Comment (stefpiatek) [inline src/components/results-list/ResultsDataFetcher.tsx:92]: We could make this a bit more clear by `?` and also using !(with && and && linking). so maybe something like this?
```suggestion
      if (!(
        patient.id &&
        patient.identifier?.get(0)?.value &&
        patient.name?.get(0)?.given &&
        patient.name?.get(0)?.family
      )) {
```

File: src/components/results-list/ResultsDataFetcher.tsx [hunk 2]
```diff
@@ -100,24 +114,42 @@ const ResultsDataFetcher: FC = () => {
...
-        if (!observation || !observation.component) {
+        if (!observation || !observation?.component || !observation?.id) {
           return;
         }
 
         const TrimmedObservation = observation.component.filter((component) => {
-          if (component?.code?.coding?.[0].code === HGVS_CDNA_LOINC) {
-            return component?.valueCodeableConcept?.coding?.[0].display;
+          if (
+            !component ||
```
Comment (stefpiatek) [inline src/components/results-list/ResultsDataFetcher.tsx:123]: I don't think this line is necessary in the guard as it's covered below

**Review by stefpiatek — APPROVED** [review]
🎉 thanks for doing this - not sure if my comment on guard clauses is that astounding but does save a small bit I hope

**Comment by jhughes982** [comment]
Thanks for the suggestions. I think that is now a bit leaner and more semantically correct. Needs a bit of concentration to follow through the logic though unfortunately.

**Pull request merged**